tests/test_fips.py

```diff
@@ -139,22 +139,50 @@ def openssl_fips_hashes_test_fnct(container_per_test: ContainerData) -> None:
     all non-fips hash algorithms fail.
 
     """
+    c = container_per_test.connection
     for digest in NONFIPS_DIGESTS:
-        cmd = container_per_test.connection.run(f"openssl {digest} /dev/null")
+        cmd = c.run(f"openssl {digest} /dev/null")
         assert cmd.rc != 0
         assert (
             "is not a known digest" in cmd.stderr
             or "Error setting digest" in cmd.stderr
         )
 
     for digest in FIPS_DIGESTS:
-        dev_null_digest = container_per_test.connection.check_output(
-            f"openssl {digest} /dev/null"
-        )
+        dev_null_digest = c.check_output(f"openssl {digest} /dev/null")
         assert f"= {NULL_DIGESTS[digest]}" in dev_null_digest, (
             f"unexpected digest of hash {digest}: {dev_null_digest}"
         )
 
+    if OS_VERSION == "15.6":
```

Comment:
I would move this to a new function that is only executed on SLES15-SP6.

Like `openssl_fips_hashes_test_fnct_old_openssl` or something similar.